src/Ideal.jl

```diff
@@ -4,17 +4,33 @@
 #
 ###############################################################################
 
-function ideal
-end
+# We assume that the functions
+#   ideal(R::Ring, x::RingElement)
+#   ideal(R::Ring, xs::AbstractVector)
+# are implemented by anyone implementing ideals for AbstractAlgebra rings.
+# The functions in this file extend the interface for `ideal`.
 
-function *(x::RingElement, R::Ring)
+function *(R::Ring, x::Any)
   return ideal(R, x)
 end
 
-function *(R::Ring, x::RingElement)
+function *(x::Any, R::Ring)
   return ideal(R, x)
 end
 
+function ideal(R::Ring, x::Any)
+  return ideal(R, R(x))
```

Comment:
I just wanted to merge this PR but then realized there are two potential issues with this method for rings which don't actually implement ideals:
1. if `x` is an element of `R`, then with this method `ideal(R,x)` results in an infinite recursion.
2. if `x` is a vector, then `R(x)` may either result in an somewhat unexpected error, or (perhaps worse) work in an unexpected way

To deal with this, I suggest we restrict this method to `x::RingElement` (which includes `x` being `Base.Integer); moreover, I would suggest we designate `ideal(R::T, xs::Vector{U})` with `U === elem_type(T)` as the central API everyone should implement (so also adjust the comment above) and delegate accordingly (of course implementations are free to implement more, e.g. to deal with principal ideals in an optima way).

Then we can just do this here (note the use of `elem_type(R)` to ensure we really delegate to `Vector
```suggestion
function ideal(R::Ring, x::RingElement)
  return ideal(R, elem_type(R)[R(x)])
```

We could then also add this to hopefully complete the delegations:
```
function ideal(R::Ring, xs::AbstractVector{T}) where T<:RingElement
  xs isa Vector{elem_type(R)} && error("ideals unsupported for ring $R")
  return ideal(R, elem_type(R)[R(x) for x in xs])
end
```